src/orchestrator.py

```
@@ -0,0 +1,247 @@
+        print(f"Loaded configuration from {config_file}")
+        return config
+    
+    def pull_image(self, image: str) -> None:
+        """Pull a Docker image from the registry."""
+        print(f"  Pulling image: {image}")
+        try:
+            self.client.images.pull(image)
+            print(f"Successfully pulled {image}")
+        except docker.errors.ImageNotFound:
+            print(f"  Image not found: {image}, will try to use local image if available")
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)